File: backend/src/main/java/org/course/ecommerce/dto/CategoryDto.java
package org.course.ecommerce.dto;

public class CategoryDto {

    private Integer categoryID;
    private String categoryName;
    private String description;

    public CategoryDto() {}

    public CategoryDto(String categoryName, String description) {
        this.categoryName = categoryName;
        this.description = description;
    }

    public Integer getCategoryID() { return categoryID; }
    public void setCategoryID(Integer categoryID) { this.categoryID = categoryID; }

    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
}

File: mobile/app/src/main/java/org/course/mobile/data/network/dto/CategoryDto.kt
package org.course.mobile.data.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// Mirrors org.course.ecommerce.dto.CategoryDto. Unlike Product/Order/OrderDetails/
// Payment/Employee (which all use "xId"), this one's id is "categoryID" (capital
// ID), so @SerialName pins it explicitly rather than relying on a naming coincidence.
@Serializable
data class CategoryDto(
    @SerialName("categoryID") val categoryId: Int? = null,
    val categoryName: String,
    val description: String? = null,
)
